backend/hitas/utils.py

```diff
@@ -78,3 +79,18 @@ def lookup_model_id_by_uuid(lookup_id: str, model_class: type[models.Model], **k
         return model_class.objects.only("id").get(uuid=uuid, **kwargs).id
     except model_class.DoesNotExist:
         raise HitasModelNotFound(model=model_class)
+
+
+TModel = TypeVar("TModel", bound=models.Model)
+
+
+def get_many_or_cached_prefetch(
+    instance: models.Model,
+    field_name: str,
+    # Optional argument to provide return type hints
+    related_model: type[TModel] = ModelBase,  # noqa
+) -> models.QuerySet[TModel]:
+    if hasattr(instance, "_prefetched_objects_cache") and field_name in instance._prefetched_objects_cache:
+        return instance._prefetched_objects_cache[field_name]
+    else:
+        return getattr(instance, field_name).all()
```

Comment:
This function only matters when the prefetched queryset has deferred fields, and we need to access the prefetched items, e.g., in the loop. Ownerships are accessed from `apartment.ownerships`, but this is a ModelManager class that cannot be iterated, so we need to evaluate a queryset from it. Usually, you would do `apartment.ownerships.all()`, but this will now create another query since the prefetched values had deferred fields, and therefore the queryset in the prefetch cache cannot be used. This function uses the prefetched queryset if one is found and then uses `.all()` if the cache is not found. If the prefetch cache has cached a queryset matching `all()`, then prefetch cache can be used, and the number of queries is the same, and thus this function is unnecessary, as in your example.

Reply:
Actually, I tested this and looked in a bit deeper into the django codebase, and it seem like [`.all()` does actually use the prefetched queryset](https://github.com/django/django/blob/main/django/db/models/fields/related_descriptors.py#L1082). In my testing last week it didn't work for some reason, but I may have been simply missing some fields in the `.only()` definition. I'll remove this method and replace them with `.all()` calls.